polaris/curation/viz_utils.py

```diff
@@ -102,3 +140,200 @@ def check_undefined_stereocenters(dataset: PandasDataFrame):
     )
     fig = dm.to_image(to_plot.mol.tolist(), legends=legends)
     return fig
+
+
+def visualize_chemspace(
+    data: pd.DataFrame,
+    mol_col: str = "smiles",
+    size_col=None,
+    seed=1428,
+    umap_metric="jaccard",
```

Comment:
I would replace by `**umap_kwargs`. Other parameters are also very influential and user will want to play with them! See e.g. https://umap-learn.readthedocs.io/en/latest/parameters.html